src/main/scala/io/iohk/ethereum/ledger/package.scala

```diff
@@ -0,0 +1,49 @@
+package io.iohk.ethereum
+
+import io.iohk.ethereum.db.storage.{EvmCodeStorage, NodeStorage}
+import io.iohk.ethereum.db.storage.EvmCodeStorage.{Code, CodeHash}
+import io.iohk.ethereum.db.storage.NodeStorage.{NodeEncoded, NodeHash}
+import io.iohk.ethereum.mpt.MerklePatriciaTrie
+import scala.language.implicitConversions
+
+package object ledger {
+
+  object MPTWrappedMap {
+    implicit def wrappedMpt[K, V](mpt: MerklePatriciaTrie[K, V]): MPTWrappedMap[K, V] = new MPTWrappedMap[K, V](mpt)
+  }
+
+  class MPTWrappedMap[K, V](mpt: MerklePatriciaTrie[K, V]) extends WrappedMap[K, V, MerklePatriciaTrie[K, V]] {
+    override val wrapped: MerklePatriciaTrie[K, V] = mpt
+
+    override def get(key: K): Option[V] = mpt.get(key)
+
+    override def update(toRemove: Seq[K], toUpsert: Seq[(K, V)]): WrappedMap[K, V, MerklePatriciaTrie[K, V]] =
+      new MPTWrappedMap[K, V](mpt.update(toRemove, toUpsert))
+  }
+
+  object EvmCodeStorageWrappedMap {
+    implicit def wrappedEmvCodeStorage[K, V](storage: EvmCodeStorage): EvmCodeStorageWrappedMap = new EvmCodeStorageWrappedMap(storage)
+  }
+
+  class EvmCodeStorageWrappedMap(storage: EvmCodeStorage) extends WrappedMap[CodeHash, Code, EvmCodeStorage] {
+    override val wrapped: EvmCodeStorage = storage
```

Comment:
similar to https://github.com/input-output-hk/etc-client/pull/72/files#diff-740c05f8806ea161939216e588eebe42R16 ?